src/app/dynamic-forms/fields/child-form/child-form.component.ts

```diff
@@ -36,6 +36,7 @@ export class ChildFormComponent {
 
 
   ngOnInit(): void {
+    const childFormLinkName = this.childFormData.childForm?.childFormLinkId?.split("].[")[0]?.replace(/[\[\]]+/g, "");
```

Comment:
If you ran the replace first and then just split by "." it might be a bit cleaner, since you wouldn't need to worry about residuals

Reply:
Might also be beneficial to add a comment here about why we only care about the first segment (per our zoom discussion)

Reply:
Moved this into a utility function mentioned in another comment